### DISCUSSION TITLE
org.apache.kyuubi.KyuubiSQLException: Invalid SessionHandle when multiple Kyuubi server is running

### DISCUSSION THREAD

**Discussion by yaroslavambrozyak** [apache/kyuubi#3413] [Q&A] [answered]
Hi, I get **Invalid SessionHandle** error when I try to call REST API to execute the query or get the result. I am running Kyuubi on k8s and get the error only if I run multiple replicas, it does not appear if I run a single instance of Kyuubi server.
I noticed that error appears if a session is created on the first server but the request to execute a query goes to the second one.
I am using Kyuubi v.1.6.0 (master branch), using an external zookeeper. Do I miss some configuration?
Thanks

**Comment by pan3793** [comment, marked as answer]
The shared state does not complete yet, you may need to use the client IP hash-based LB strategy to make sure one client always accesses the same Kyuubi instance for now.

**Reply by yaroslavambrozyak** [reply]
Got it, thanks for a quick reply

**Comment by yaroslavambrozyak** [comment]
Hi @pan3793, just want to check, is there any progress on this feature? Do you have an approx ETA? We are planning to have multiple Kyuubi instances in production

**Comment by v1gn3shwar** [comment]
Any update on this thread?

**Reply by pan3793** [reply]
AFAIK, the existing core developers do not put it in high priority, and it's an open-source and community-driven project, if anyone highly requires this feature, please feel free to send PR, I am happy to help review it.
